python/llm/src/bigdl/llm/transformers/models/baichuan.py

```diff
@@ -70,6 +70,8 @@ def baichuan_attention_forward_7b(
         cache_k = past_key_value[0]
         cache_v = past_key_value[1]
         if cache_k.stride()[1] <= cache_k.size(2) * cache_k.size(3):
+            if device.type == 'xpu':
+                torch.xpu.empty_cache()
```

Comment:
better to have two versions: `init_kv_cache` and `extend_kv_cache`

Reply:
I will try such refactor, we can merge this pr first to better user experience.